Source/KerbalConstructionTime/Utilities/Utilities.cs

```diff
@@ -2122,6 +2126,131 @@ public static void CleanupDebris(string launchSiteName)
+            var lockedParts = purchasableParts.Where(kvp => kvp.Value.Status == PurchasabilityStatus.Unavailable).Select(kvp => kvp.Key).ToList();
+            List<string> pendingTech = new List<string>();
+
+            foreach (AvailablePart p in lockedParts)
+            {
+                if (!pendingTech.Contains(p.TechRequired)) pendingTech.Add(p.TechRequired);
+            }
+
+            // now back through the list looking for upgrades to add to our batch list
+            foreach (Part p in ship.Parts)
+            {
+                var mecs = p.FindModulesImplementing<RealFuels.ModuleEngineConfigsBase>();
+                foreach (var mec in mecs)
+                {
+                    var sanitizedName = RealFuels.Utilities.SanitizeName(mec.configuration);
+                    if (ecmPartsList.Contains(sanitizedName))
+                        continue;
+                    
+                    ecmPartsList.Add(sanitizedName);
+                    ConfigNode n = mec.configs.Find(c => c.GetValue("name") == mec.configuration);
+                    if (n != null)
+                    {
+                        string techRequired = n.GetValue("techRequired");
+                        if (techRequired == null) continue;
+
+                        ProtoTechNode techState = ResearchAndDevelopment.Instance.GetTechState(techRequired);
+                        bool techIsUnlocked = techState != null && techState.state == RDTech.State.Available;
+                        if (!techIsUnlocked && !pendingTech.Contains(techRequired)) 
+                            pendingTech.Add(techRequired);
+                    }
+                }
+            }
+
+            var ecmCost = RealFuels.EntryCostManager.Instance.ConfigEntryCost(ecmPartsList);
+
+            var techList = RemoveParentsFromTechList(pendingTech);
+            float totalCost = runningCost + Convert.ToSingle(ecmCost);
+            KCTDebug.Log($"Vessel parts unlock cost check. Total: {totalCost}, Raw cost: {runningCost}, ECM cost: {ecmCost}");
+            return new Tuple<float, List<string>>(totalCost, techList);
+        }
+
+        public static List<string> RemoveParentsFromTechList(List<string> input)
+        {
+            List<string> blacklist = new List<string>();
+            SortedList<string,string> slist = new SortedList<string, string>();
+            foreach(string s in input)
+            {
+                foreach (string parent in KerbalConstructionTimeData.techNameToParents[s])
+                {
+                    if (!blacklist.Contains(parent)) blacklist.Add(parent);
+                }
+            }
+            foreach (string s in input)
+            {
+                if (!blacklist.Contains(s))
+                {
+                    // sort our result, depth into the tree then alpha
+                    int depth = KerbalConstructionTimeData.techNameToParents[s].Count();
+                    string skey = $"{depth:d2}{s}";
```

Comment:
Since we're already using LINQ pretty much everywhere, maybe `OrderBy().ThenBy()`?

Reply:
I could make it a list of Tuple<int,string>, put those in a list, and then return that with the OrderBy().ThenBy(), but I don't see that it would be any better. The int for depth is not going to be unique. We could just do a pure alpha order rather than trying to use depth as a proxy for lateness in the tree.

Reply:
I mean, if you deliberately tried to maximize it, you might get as many as a dozen entries in this list, and the sort is only going to need to get a few characters deep into most names to figure out position. Optimizing that seems kind of pointless.